### PR TITLE
DOC: Added examples to pandas.concat documentation

### PR DESCRIPTION
**Pull request by palewire**
I've added two simple examples to the ``pd.concat`` documentation. I've also done a very small amount of clean up on the rest of the docstring.

 - [x] passes ``git diff upstream/master | flake8 --diff``

### CODE REVIEW

File: pandas/tools/merge.py [hunk 1]
```diff
@@ -1436,13 +1438,51 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
-    Notes
-    -----
-    The keys, levels, and names arguments are all optional
-
     Returns
     -------
     concatenated : type of objects
+
+    Notes
+    -----
```
Comment (jreback) [inline pandas/tools/merge.py]: if you are going to add examples, then this needs to be more complete, using examples from here:http://pandas.pydata.org/pandas-docs/stable/merging.html#concatenating-objects

certainly not all of these, but a quick selection (as well as having this link would be fine).

**Comment by palewire** [comment]
That's fine by me. I'll add a link to that walkthrough and a wider array of
examples. I'll aim for the new examples to be more discrete and less
holistic than the walkthrough.

File: pandas/tools/merge.py [hunk 2]
```diff
@@ -1436,13 +1438,51 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    The keys, levels, and names arguments are all optional.
+
+    Examples
+    --------
+    Combine two ``Series``.
+
+        >>> import pandas as pd
+        >>> s1 = pd.Series(['a', 'b'])
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: no indentation is needed here for the example blocks

(and the 'import pandas as pd' can also be left out -> this is assumed)

File: pandas/tools/merge.py [hunk 3]
```diff
@@ -705,7 +705,7 @@ def _get_join_info(self):
                 _left_join_on_index(left_ax, right_ax, self.left_join_keys,
                                     sort=self.sort)
 
-        elif self.left_index and self.how == 'right':
+        elif self.tleft_index and self.how == 'right':
```
Comment (jreback) [inline pandas/tools/merge.py]: pls don't change code in the same PR as docs (unless its germane), I assume this was a typo though.

File: pandas/tools/merge.py [hunk 4]
```diff
@@ -1436,13 +1438,90 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    >>> s1 = pd.Series(['a', 'b'])
+    >>> s2 = pd.Series(['c', 'd'])
+    >>> pd.concat([s1, s2], ignore_index=True)
+    0    a
+    1    b
+    2    c
+    3    d
+    dtype: object
+
+    Add a ``hierarchical index`` at the outermost level of
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: no need to add backtick quotes around 'hierarchical index'

File: pandas/tools/merge.py [hunk 5]
```diff
@@ -1436,13 +1438,90 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    >>> c = pd.concat([s1, s2], keys=["s1", 's2',])
+    >>> c
+    s1  0    a
+        1    b
+        2    c
+    s2  0    c
+        1    d
+        2    e
+    dtype: object
+    >>> c.ix['s1']
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: The above example is good. But I would leave out the examples on indexing the result to keep the examples focused on `concat` (such examples would certainly be nice for the tutorial docs)

Maybe add an example with `axis=1`

File: pandas/tools/merge.py [hunk 6]
```diff
@@ -1436,13 +1438,90 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    2    c
+    3    d
+    dtype: object
+
+    Add a ``hierarchical index`` at the outermost level of
+    the data.
+
+    >>> s1 = pd.Series(['a', 'b', 'c'])
+    >>> s2 = pd.Series(['c', 'd', 'e'])
+    >>> c = pd.concat([s1, s2], keys=["s1", 's2',])
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: can you use consistent quotes within the keys list (eg always single quotes)?

**Review by jorisvandenbossche — COMMENTED** [review]
Nice examples! Added some more comments

File: pandas/tools/merge.py (hunk 4 above)
Comment (palewire) [inline pandas/tools/merge.py]: I've fixed this.

File: pandas/tools/merge.py (hunk 6 above)
Comment (palewire) [inline pandas/tools/merge.py]: I've fixed this.

File: pandas/tools/merge.py (hunk 5 above)
Comment (palewire) [inline pandas/tools/merge.py]: I've removed the `ix` calls and added axis=1 to my list for later.

File: pandas/tools/merge.py (hunk 3 above)
Comment (palewire) [inline pandas/tools/merge.py]: It was. I'm sorry for the oversight.

**Comment by palewire** [comment]
@jorisvandenbossche and @jreback, I believe I've made the changes you've asked for. Please let me know if there's anything more I can do.

File: pandas/tools/merge.py [hunk 7]
```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    1    b
+    0    c
+    1    d
+    dtype: object
+
+    Clear the existing index and reset it in the result
+    by setting the ``ignore_index`` option to ``True``.
+
+    >>> s1 = pd.Series(['a', 'b'])
+    >>> s2 = pd.Series(['c', 'd'])
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: Can you leave out the redefinition of s1 and s2 ? (to save some vertical space) We can assume that the example build on each other.

File: pandas/tools/merge.py [hunk 8]
```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    1    b
+    2    c
+    3    d
+    dtype: object
+
+    Add a hierarchical index at the outermost level of
+    the data with the ``keys`` option.
+
+    >>> s1 = pd.Series(['a', 'b', 'c'])
+    >>> s2 = pd.Series(['c', 'd', 'e'])
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: Here as well (you can use the same s1 and s2 I think)

File: pandas/tools/merge.py [hunk 9]
```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+
+    Label the index keys you create with the ``names`` option.
+
+    >>> s1 = pd.Series(['a', 'b', 'c'])
+    >>> s2 = pd.Series(['c', 'd', 'e'])
+    >>> pd.concat(
+    ...     [s1, s2],
+    ...     keys=['s1', 's2',],
+    ...     names=['Series name', 'Row ID']
+    ... )
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: Style-wise, can you do 

```
pd.concat([s1, s2], keys=[...],
          names=[...])
```

to have a more consistent style within the docs

File: pandas/tools/merge.py [hunk 10]
```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+                 1         d
+                 2         e
+    dtype: object
+
+    Combine two ``DataFrame`` objects with identical columns.
+
+    >>> df1 = pd.DataFrame(
+    ...     [['a', 1], ['b', 2]],
+    ...     columns=['letter', 'number']
+    ... )
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: the same here regarding style

File: pandas/tools/merge.py [hunk 11]
```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    >>> df1
+      letter  number
+    0      a       1
+    1      b       2
+    >>> df2 = pd.DataFrame(
+    ...     [['c', 3, 'cat'], ['d', 4, 'dog']],
+    ...     columns=['letter', 'number', 'animal']
+    ... )
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: also here, I would leave out the redefinition and display of the same dataframes (then what is shown is more focused on the result)

Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: I meant this comment actually for the example below, as here df2 is actually changed (maybe only show df2 creation and display in this example)

File: pandas/tools/merge.py [hunk 12]
```diff
@@ -1436,13 +1438,198 @@ def concat(objs, axis=0, join='outer', join_axes=None, ignore_index=False,
...
+    ...     [['a', 1], ['b', 2]],
+    ...     columns=['letter', 'number']
+    ... )
+    >>> df1
+      letter  number
+    0      a       1
+    1      b       2
+    >>> df2 = pd.DataFrame(
+    ...     [['c', 3], ['d', 4]],
+    ...     columns=['letter', 'number']
```
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: I would use here eg the 'animal' column name to have different columns (that is IMO a more natural example to concat columns with different names)

**Review by jorisvandenbossche — COMMENTED** [review]
Some stylistic comments on the examples you added.

File: pandas/tools/merge.py (hunk 7 above)
Comment (palewire) [inline pandas/tools/merge.py]: I will make this change and push it shortly.

File: pandas/tools/merge.py (hunk 8 above)
Comment (palewire) [inline pandas/tools/merge.py]: Change coming.

File: pandas/tools/merge.py (hunk 9 above)
Comment (palewire) [inline pandas/tools/merge.py]: Will do. I was unaware that was the established style.

File: pandas/tools/merge.py (hunk 11 above)
Comment (palewire) [inline pandas/tools/merge.py]: Will do.

File: pandas/tools/merge.py (hunk 12 above)
Comment (palewire) [inline pandas/tools/merge.py]: Will do.

**Comment by palewire** [comment]
@jorisvandenbossche I believe I have made the changes you've requested. Please let me know if there's more I can do.

File: pandas/tools/merge.py (hunk 9 above)
Comment (jorisvandenbossche) [inline pandas/tools/merge.py]: No, no problem. We don't have this documented clearly what is expected here (and are not always that consistent in existing docs)

**Comment by jorisvandenbossche** [comment]
@palewire Thanks!

**Pull request merged**